### ISSUE TITLE
No IDE roles found on  System : HXE2.0

### ISSUE THREAD

**Issue by aj-code-synch** [SAPDocuments/Tutorials#1477]
Hello,
     I am running HANA Express 2.0 on Amazon cloud. I installed the Server + Applications. I can log in via SYSTEM and XSA_DEV. XSA_DEV does not have any ide roles on it. I tried assigning the roles via SYSTEM user login but sap.hana.xs.ide.roles::Developer roles don't exist on the system. Any pointer for this?

**Comment by Lsubatin** [comment]
Hi, XSA_DEV and XSA_ADMIN should have role collections as explained here (they are granted in HXE by default): https://blogs.sap.com/2016/08/16/sap-web-ide-for-sap-hana-sps-12-patch-1-tips-and-tricks/  

You won't see them in HANA Studio but they  provide access to the Web IDE  (https://host:53075/) 
Otherwise, I have an HXE 2.0 with XSA and see the sap.hana.xs.ide.roles::Developer role is there.

**Comment by aj-code-synch** [comment]
I have tried many times to install HANA Express on AWS but every time there is something wrong. I have tried with HXE1.0. It did create the roles (I was able to see them in the studio). But for HXE1.0 the webide URL did not work. 

I tried HXE2.0 and the roles don't get generated. The XSA installation fails in between. I tried patching it via hdblcm but it's not 100%. 

The installation on AWS has been bit iffy throughout. I could always go with CAL set up but by default, it picks up an instance type which is a bit costly for a personal sandbox in my opinion.  Is there any guide on how to install the HXE on AWS manually?

**Comment by Lsubatin** [comment]
There is a guide on how to install HXE in the Google Cloud Platform:  https://blogs.sap.com/2017/02/07/hana-express-edition-on-google-cloud-platform-and-ubuntu/ 

i do not have the step by step for amazon updated yet. Whatever platform you choose, remember to add more resources to the machine before starting the installation so you do not get timeout errors.

**Comment by aj-code-synch** [comment]
Thanks! I took a quick look at your guide for Google Cloud and it's interesting that you used Ubuntu instead of SLES.  Does the SAP CAL on Google Cloud also use Ubuntu? 

I will probably give Google cloud a try once I give up on Amazon!

**Comment by Lsubatin** [comment]
Hi, the only reason why I used Ubuntu is that it won't charge for the license. The supported operating systems for SAP HANA are only versions SLES and RHEL, so there are no official images with other OSs, HXE is a community- supported model so it's not like the community will withdraw support from you if you use other operating systems. The GCP Launcher version uses SLES.  Cheers!

**Comment by aj-code-synch** [comment]
Thanks! I reinstalled the HXE2.0 on AWS and used hdblcm instead of setup script and it went fine. No errors and XSa was installed fine. However, the ide roles were not created. I ran the optimize script but it was not able to create the roles because of some authorization issues. I ran the script as hxeadm.  Is there another way the roles can be created?

**Comment by aj-code-synch** [comment]
I ran it as HXEADM. It does not allow it to be run as EC2-USER and asks me to login as HXEADM to run it. I am attaching the result in txt format. 
[optimize_results.txt](https://github.com/SAPDocuments/Tutorials/files/917849/optimize_results.txt)

**Comment by Lsubatin** [comment]
I see it's an authentication error (as if using the wrong password). What happens if you try logging in manually using _hdbsql -u SYSTEM  -i 90 -d SystemDB -n <<the hostname you entrered when prompted by the installation script>>_  ?  

If it lets you in, can you create the XSA_DEV, XSA_SHINE and TEL_ADMIN users (create user <<user name>> password <<some temp password>>) and run the script again? If all of that works, you can assign the roles manually by logging in to xs as XSA_ADMIN (use command: xs login ) and then use command xs set-space-role (more details here: https://www.sap.com/developer/how-tos/2016/09/hxe-howto-tutorialprep.html ) 

You can also assign those users (once created) to the roles and spaces at https://host:51015, entering as XSA_ADMIN which is apparently all set up judging from that log.

**Comment by aj-code-synch** [comment]
Thanks for your help on this. I was able to login via SYSTEM user and created the users manually. I ran the script again but it fails giving an error that roles that it's trying to create have invalid names. 

The script added the developer space role to XSA_DEV but on WebIDe I get a forbidden error. 
[optimize_results2.txt](https://github.com/SAPDocuments/Tutorials/files/920223/optimize_results2.txt)


` ./hxe_optimize.sh
Enter HANA instance number [90]:
Enter "System database user (SYSTEM)" password :
Enter new password for "Telemetry technical user (TEL_ADMIN)":
Enter new confirm password for "Telemetry technical user (TEL_ADMIN)":

Enter "XSA administrative user (XSA_ADMIN)" password :
Enter new password for "XSA development user (XSA_DEV)":
Enter new confirm password for "XSA development user (XSA_DEV)":
Enter new password for "XSA Shine application user (XSA_SHINE)":
Enter new confirm password for "XSA Shine application user (XSA_SHINE)":
Login to XSA services...

API_URL: https://ec2-54-87-254-119.compute-1.amazonaws.com:39030
USERNAME: xsa_admin
Authenticating...
ORG: HANAExpress
SPACE: SAP
API endpoint:   https://ec2-54-87-254-119.compute-1.amazonaws.com:39030 (API version: 1)
User:           xsa_admin
Org:            HANAExpress
Space:          SAP

Check/Wait for all apps to start...

Waiting for apps: auditlog-server, auditlog-broker, deploy-service, auditlog-odata, auditlog-ui, product-installer, hrtt-service, sqlanlz-sv  c, sqlanlz-ui, hrtt-core, xsa-admin-backend, xsa-admin, di-local-npm-registry, di-core, devx-ui5, di-runner, webide, sapui5_fesv2, cockpit-p  ersistence-svc, cockpit-hdb-svc, cockpit-collection-svc, cockpit-hdbui-svc, cockpit-landscape-svc, cockpit-web-app, cockpit-adminui-svc, coc  kpit-admin-web-app
   auditlog-server: ready
   auditlog-broker: ready
   deploy-service: ready
   auditlog-odata: ready
   auditlog-ui: ready
   product-installer: ready
   hrtt-service: ready
   sqlanlz-svc: ready
   sqlanlz-ui: ready
   hrtt-core: ready
   xsa-admin-backend: ready
   xsa-admin: ready
   di-local-npm-registry: ready
   di-core: ready
   devx-ui5: ready
   di-runner: ready
   webide: ready
   sapui5_fesv2: ready
   cockpit-persistence-svc: ready
   cockpit-hdb-svc: ready
   cockpit-collection-svc: ready
   cockpit-hdbui-svc: ready
   cockpit-landscape-svc: ready
   cockpit-web-app: ready
   cockpit-adminui-svc: ready
   cockpit-admin-web-app: ready
OK

Create telemetry technical user "TEL_ADMIN" on "SystemDB" database...
* 331: user name already exists: TEL_ADMIN: line 1 col 13 (at pos 12) SQLSTATE: HY000
0 rows affected (overall time 2125 usec; server time 1312 usec)

0 rows affected (overall time 34.799 msec; server time 15.722 msec)

0 rows affected (overall time 218.891 msec; server time 196.399 msec)

0 rows affected (overall time 2109 usec; server time 1265 usec)

Create telemetry technical user "TEL_ADMIN" on "HXE" database...
0 rows affected (overall time 19.345 msec; server time 18.535 msec)

0 rows affected (overall time 2092 usec; server time 1363 usec)

0 rows affected (overall time 41.051 msec; server time 21.824 msec)

0 rows affected (overall time 222.296 msec; server time 199.882 msec)

0 rows affected (overall time 2059 usec; server time 1281 usec)

Change telemetry URL for "SystemDB" database...


The Telemetry Server URL was successfully updated to https://telemetry.cloud.sap for the database, SystemDB.

Change telemetry URL for "HXE" database...


The Telemetry Server URL was successfully updated to https://telemetry.cloud.sap for the database, HXE.

0 rows affected (overall time 19.692460 sec; server time 19.691247 sec)

0 rows affected (overall time 83.835 msec; server time 82.492 msec)

0 rows affected (overall time 62.579 msec; server time 61.318 msec)

0 rows affected (overall time 99.829 msec; server time 98.933 msec)

0 rows affected (overall time 81.399 msec; server time 80.345 msec)

0 rows affected (overall time 61.878 msec; server time 60.881 msec)

0 rows affected (overall time 63.448 msec; server time 62.567 msec)

0 rows affected (overall time 72.529 msec; server time 71.663 msec)

0 rows affected (overall time 60.606 msec; server time 59.770 msec)

0 rows affected (overall time 7865 usec; server time 6831 usec)

0 rows affected (overall time 8904 usec; server time 7731 usec)

0 rows affected (overall time 58.523 msec; server time 57.703 msec)

0 rows affected (overall time 62.746 msec; server time 61.519 msec)

0 rows affected (overall time 15.088 msec; server time 14.334 msec)

Creating the XSA_ADMIN users and space...

API_URL: https://ec2-54-87-254-119.compute-1.amazonaws.com:39030
USERNAME: xsa_admin
Authenticating...
ORG: HANAExpress
SPACE: SAP
API endpoint:   https://ec2-54-87-254-119.compute-1.amazonaws.com:39030 (API version: 1)
User:           xsa_admin
Org:            HANAExpress
Space:          SAP

Create role collections for WEB_IDE...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  5340    0  5340    0     0  23556      0 --:--:-- --:--:-- --:--:-- 23628
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   498    0   498    0     0   9732      0 --:--:-- --:--:-- --:--:--  9764
0 rows affected (overall time 60.323 msec; server time 59.469 msec)

0 rows affected (overall time 58.740 msec; server time 57.946 msec)

Create the xsa_dev user...
* 331: user name already exists: XSA_DEV: line 1 col 13 (at pos 12) SQLSTATE: HY000
0 rows affected (overall time 17.428 msec; server time 16.624 msec)


Updating space "SAP" in org "HANAExpress"...
OK
TIP: Use 'xs restart' on apps in this space to ensure the changes take effect


Adding role 'SpaceDeveloper' to user XSA_ADMIN in space "SAP" of org "HANAExpress" ...
FAILED: User with UAA id '156765' (guid '5ca902ca-7a81-4ba0-88dc-4ba20761a4e8') already has role 'SpaceDeveloper' in space "SAP".

Adding role 'SpaceDeveloper' to user XSA_DEV in space "SAP" of org "HANAExpress" ...
OK

Creating and configuring the XSA_SHINE user for the HANA Shine application...
* 331: user name already exists: XSA_SHINE: line 1 col 13 (at pos 12) SQLSTATE: HY000

Adding role 'SpaceDeveloper' to user XSA_SHINE in space "SAP" of org "HANAExpress" ...
OK


Adding role 'OrgManager' to user XSA_ADMIN in org "HANAExpress" ...
FAILED: User with UAA id '156765' (guid '5ca902ca-7a81-4ba0-88dc-4ba20761a4e8') already has role 'OrgManager' in organization "HANAExpress".
Deploy di-builder for WEB_IDE...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  5255    0  5255    0     0  22962      0 --:--:-- --:--:-- --:--:-- 23048
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   109    0   109    0     0    160      0 --:--:-- --:--:-- --:--:--   160
di-core has started
Deploying builder to space SAP with id bea3046c-48bf-41c6-bfe2-546e1d6c543a
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   141  100   127  100    14    882     97 --:--:-- --:--:-- --:--:--   888
Builder deployment started
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   126  100   126    0     0   1126      0 --:--:-- --:--:-- --:--:--  1135
Successfully deployed builder in space SAP
Stop applications...

API_URL: https://ec2-54-87-254-119.compute-1.amazonaws.com:39030
USERNAME: xsa_admin
Authenticating...
ORG: HANAExpress
SPACE: SAP
API endpoint:   https://ec2-54-87-254-119.compute-1.amazonaws.com:39030 (API version: 1)
User:           xsa_admin
Org:            HANAExpress
Space:          SAP

Processing: jobscheduler-backend

Showing status and information about "jobscheduler-backend"
  name:              jobscheduler-backend
  requested state:   STOPPED
  instances:         1
  memory:            1.00 GB
  disk:              <unlimited>
  buildpack:         <default>
  urls:              https://ec2-54-87-254-119.compute-1.amazonaws.com:51029

Processing: jobscheduler-broker

Showing status and information about "jobscheduler-broker"
  name:              jobscheduler-broker
  requested state:   STOPPED
  instances:         1
  memory:            1.00 GB
  disk:              <unlimited>
  buildpack:         <default>
  urls:              https://ec2-54-87-254-119.compute-1.amazonaws.com:51028

Processing: jobscheduler-dashboard

Showing status and information about "jobscheduler-dashboard"
  name:              jobscheduler-dashboard
  requested state:   STOPPED
  instances:         1
  memory:            1.00 GB
  disk:              <unlimited>
  buildpack:         <default>
  urls:              https://ec2-54-87-254-119.compute-1.amazonaws.com:51030

Processing: jobscheduler-db

Showing status and information about "jobscheduler-db"
  name:              jobscheduler-db
  requested state:   STOPPED
  instances:         1
  memory:            256 MB
  disk:              <unlimited>
  buildpack:         <default>
  urls:              <none>

Processing: jobscheduler-rest

Showing status and information about "jobscheduler-rest"
  name:              jobscheduler-rest
  requested state:   STOPPED
  instances:         1
  memory:            1.00 GB
  disk:              <unlimited>
  buildpack:         <default>
  urls:              https://ec2-54-87-254-119.compute-1.amazonaws.com:51026

Processing: jobscheduler-service

Showing status and information about "jobscheduler-service"
  name:              jobscheduler-service
  requested state:   STOPPED
  instances:         1
  memory:            1.00 GB
  disk:              <unlimited>
  buildpack:         <default>
  urls:              https://ec2-54-87-254-119.compute-1.amazonaws.com:51027


Showing status and information about "di-cert-admin-ui"
  name:              di-cert-admin-ui
  requested state:   STOPPED
  instances:         1
  memory:            16.0 MB
  disk:              <unlimited>
  buildpack:         <default>
  urls:              https://ec2-54-87-254-119.compute-1.amazonaws.com:51022


Showing status and information about "di-space-enablement-ui"
  name:              di-space-enablement-ui
  requested state:   STOPPED
  instances:         1
  memory:            16.0 MB
  disk:              <unlimited>
  buildpack:         <default>
  urls:              https://ec2-54-87-254-119.compute-1.amazonaws.com:51023


Showing status and information about "sap-portal-services"
  name:              sap-portal-services
  requested state:   STOPPED
  instances:         1
  memory:            256 MB
  disk:              <unlimited>
  buildpack:         <default>
  urls:              https://ec2-54-87-254-119.compute-1.amazonaws.com:51017

Logging out of XSA...

Logging out...
OK

* 389: invalid role name: sap.hana.ide.roles::EditorDeveloper SQLSTATE: HY000
* 389: invalid role name: sap.hana.ide.roles::CatalogDeveloper SQLSTATE: HY000
* 389: invalid role name: sap.hana.ide.roles::SecurityAdmin SQLSTATE: HY000
* 389: invalid role name: sap.hana.ide.roles::TraceViewer SQLSTATE: HY000
Turning on statistics server...
0 rows affected (overall time 63.853 msec; server time 63.086 msec)

Reclaiming log space...
0 rows affected (overall time 1304 usec; server time 517 usec)

Reclaiming disk space...
0 rows affected (overall time 23.986207 sec; server time 23.985421 sec)

hdbdaemon will wait maximal 300 seconds for NewDB services finishing.
Stopping instance using: /usr/sap/HXE/SYS/exe/hdb/sapcontrol -prot NI_HTTP -nr 90 -function Stop 400

13.04.2017 15:48:45
Stop
OK
Waiting for stopped instance using: /usr/sap/HXE/SYS/exe/hdb/sapcontrol -prot NI_HTTP -nr 90 -function WaitforStopped 600 2


13.04.2017 15:49:21
WaitforStopped
OK
hdbdaemon is stopped.


StartService
Impromptu CCC initialization by 'rscpCInit'.
  See SAP note 1266393.
OK
OK
Starting instance using: /usr/sap/HXE/SYS/exe/hdb/sapcontrol -prot NI_HTTP -nr 90 -function StartWait 2700 2


13.04.2017 15:49:21
Start
OK

13.04.2017 15:50:17
StartWait
OK
Register SystemDB database HDB resource with Cockpit...
0 rows affected (overall time 3926 usec; server time 1920 usec)

0 rows affected (overall time 3458 usec; server time 1717 usec)

===============================================================================
Register resource on "SystemDB" database
===============================================================================



Login to XSA services...
Check/Wait for Cockpit app to start...

Waiting for apps: cockpit-adminui-svc, cockpit-admin-web-app.................................................................
   cockpit-admin-web-app: ready
Waiting for apps: cockpit-adminui-svc
   cockpit-adminui-svc: ready
OK

Get client ID, client secret key, UAA URL, ADMINUI, ADMIN_WEB_APP URL...
Get authentication token from UAA...
Register resource...
"SystemDB" database is registered to Cockpit.
Create role colllections...
Role colllections created.
0 rows affected (overall time 8159 usec; server time 5831 usec)

0 rows affected (overall time 4180 usec; server time 1986 usec)

===============================================================================
Configure proxy
===============================================================================


No proxy to configure.
0 rows affected (overall time 1917 usec; server time 868 usec)

0 rows affected (overall time 173.716 msec; server time 172.589 msec)

0 rows affected (overall time 203.752 msec; server time 187.564 msec)

Collect garbage on "hdbnameserver"...
SAP HANA DB Management Client Console (type '\?' to get help for client commands)
Try to open connection to server process 'hdbnameserver' on system 'HXE', instance '90'
SAP HANA DB Management Server Console (type 'help' to get help for server commands)
Executable: hdbnameserver (PID: 13594)
[OK]
--
Garbage collection. Starting with 4821704704 allocated bytes.
3426185216 bytes allocated after garbage collection.
[OK]
--
[EXIT]
--
[BYE]

Collect garbage on "hdbindexserver"...
SAP HANA DB Management Client Console (type '\?' to get help for client commands)
Try to open connection to server process 'hdbindexserver' on system 'HXE', instance '90'
SAP HANA DB Management Server Console (type 'help' to get help for server commands)
Executable: hdbindexserver (PID: 13754)
[OK]
--
Garbage collection. Starting with 2554408960 allocated bytes.
2228953088 bytes allocated after garbage collection.
[OK]
--
[EXIT]
--
[BYE]

Collect garbage on "hdbcompileserver"...
SAP HANA DB Management Client Console (type '\?' to get help for client commands)
Try to open connection to server process 'hdbcompileserver' on system 'HXE', instance '90'
SAP HANA DB Management Server Console (type 'help' to get help for server commands)
Executable: hdbcompileserver (PID: 13719)
[OK]
--
Garbage collection. Starting with 1307377664 allocated bytes.
837120000 bytes allocated after garbage collection.
[OK]
--
[EXIT]
--
[BYE]

Collect garbage on "hdbpreprocessor"...
SAP HANA DB Management Client Console (type '\?' to get help for client commands)
Try to open connection to server process 'hdbpreprocessor' on system 'HXE', instance '90'
SAP HANA DB Management Server Console (type 'help' to get help for server commands)
Executable: hdbpreprocessor (PID: 13721)
[OK]
--
Garbage collection. Starting with 1499758592 allocated bytes.
1027457024 bytes allocated after garbage collection.
[OK]
--
[EXIT]
--
[BYE]

Collect garbage on "hdbdiserver"...
SAP HANA DB Management Client Console (type '\?' to get help for client commands)
Try to open connection to server process 'hdbdiserver' on system 'HXE', instance '90'
SAP HANA DB Management Server Console (type 'help' to get help for server commands)
Executable: hdbdiserver (PID: 13872)
[OK]
--
Garbage collection. Starting with 1316253696 allocated bytes.
939347968 bytes allocated after garbage collection.
[OK]
--
[EXIT]
--
[BYE]

Stop "HXE" tenant database...
0 rows affected (overall time 13.058243 sec; server time 13.056703 sec)
`

**Comment by Lsubatin** [comment]
I get the "forbidden" error when I log in using the same browser as XSA_ADMIN. Try an incognito session with XSA_DEV for the webide.  Those errors do not seem to be related to the "Forbidden".  Are the proper role collections assigned to XSA_DEV? 

![image](https://cloud.githubusercontent.com/assets/24896969/25067929/a5aa6094-2228-11e7-9b32-2179ac907e6f.png)

As for those errors, they are related to assigning those roles to the SYSTEM user. What happens if your try the following SQL?

"CALL GRANT_ACTIVATED_ROLE ('sap.hana.ide.roles::EditorDeveloper','SYSTEM')"
"CALL GRANT_ACTIVATED_ROLE ('sap.hana.ide.roles::CatalogDeveloper','SYSTEM')"
"CALL GRANT_ACTIVATED_ROLE ('sap.hana.ide.roles::SecurityAdmin','SYSTEM')"
"CALL GRANT_ACTIVATED_ROLE ('sap.hana.ide.roles::TraceViewer','SYSTEM')"

Do you see those roles in HANA Studio?

**Comment by aj-code-synch** [comment]
The ide roles were not in the list of roles in Studio. That was the main difference that I saw between HXE1.0 and HXE2.0. The role creation via optimizing script failed giving an error of invalid role names. 

I reinstalled HXE1.0 using hdblcm instead of setup file and ran the optimize script as well. So far everything is looking good and I was able to create a simple app. 

As for HXE2.0, I deleted the system as it was just causing my money and time! 

You have been really helpful on this Lucia. Thanks !
